### PR TITLE
Chore/run e2e tests from within the repo

### PR DESCRIPTION
**Pull request by tahb**
# Context

https://dsdmoj.atlassian.net/browse/CAS2-140?atlOrigin=eyJpIjoiOGYwMzUwYmJkMWZkNDY3NGEyNGUyZTM3ZGFkYmIzYjciLCJwIjoiaiJ9

It’s important to note that e2e tests are required to pass for deploying the frontend to preprod and prod.

We found that it can be difficult to coordinate the frontend changes (a pull request in 1 repo) alongside the changes to the e2e tests (a pull request in another repo).

Both must be worked on and tested locally from different repos. We also have double the admin of proposing and reverting two pull requests, rather than just one.

We’d like to move these tests from a separate repo to the same repo.

# Changes

- We install playwright and copy over the e2e tests from the separate repo.
- We add a very simple health check test to help quickly verify the e2e tests are set up correctly. We could drop this if needed?

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: playwright.config.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    {
+      name: 'setupDev',
+      testMatch: /.*\.setup\.ts/,
+      use: { baseURL: 'https://community-accommodation-tier-2-dev.hmpps.service.justice.gov.uk' },
+    },
+    {
+      name: 'dev',
+      use: {
+        ...devices['Desktop Chrome'],
+        storageState: 'e2e-tests/playwright/.auth/user.json',
```
Comment (tahb) [inline playwright.config.ts:28]: I tweaked this

File: playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import { TestOptions } from './testOptions'
+
+export default defineConfig<TestOptions>({
+  testDir: './e2e-tests/tests',
+  fullyParallel: false,
+  forbidOnly: !!process.env.CI,
+  retries: process.env.CI ? 2 : 0,
+  maxFailures: 1,
+  workers: 1,
+  reporter: [['html', { outputFolder: './e2e-tests/playwright-report/index.html' }]],
```
Comment (tahb) [inline playwright.config.ts:12]: I tweaked this

File: playwright.config.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { defineConfig, devices } from '@playwright/test'
+import 'dotenv/config'
+import { TestOptions } from './testOptions'
+
+export default defineConfig<TestOptions>({
+  testDir: './e2e-tests/tests',
```
Comment (tahb) [inline playwright.config.ts:6]: I tweaked this

File: e2e-tests/tests/auth.setup.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { test as setup } from '../test'
+
+const authFile = 'e2e-tests/playwright/.auth/user.json'
```
Comment (tahb) [inline e2e-tests/tests/auth.setup.ts:3]: I tweaked this

**Comment by tahb** [comment]
Once this is reviewed I need to go back to the e2e test repo to collect any new changes!

File: package.json [hunk 5]
```diff
@@ -34,7 +34,12 @@
     "clean": "rm -rf dist build node_modules stylesheets",
     "test:integration": "npm run start-test-wiremock && start-server-and-test start-feature http://localhost:3007/ping int-test",
     "test:integration:ui": "npm run start-test-wiremock && start-server-and-test start-feature:dev http://localhost:3007/ping int-test-ui",
-    "start-test-wiremock": "docker-compose -f docker-compose-test.yml up -d"
+    "start-test-wiremock": "docker-compose -f docker-compose-test.yml up -d",
+    "test:e2e": "npx playwright test --project=setupDev --project=dev",
+    "test:e2e:ui": "npm run test -- --ui",
```
Comment (Harriethw) [inline package.json:39]: should this be `npm run playwright test -- --ui`?

Comment (Harriethw) [reply package.json:39]: or `npm run test:e2e -- --ui` ?

**Review by Harriethw — APPROVED** [review]
nice! It seems to work for me locally and makes sense 👍

File: package.json (hunk 5 above)
Comment (patrickjfl) [reply package.json:39]: `npm run test:e2e -- --ui` works for me

**Review by patrickjfl — APPROVED** [review]
Nice job Tom! LGTM, although I think Harriet's right about the command to run them against dev, and putting something in the readme about the automatic filtering out of all tests other than auth the first time you run them in the UI.

**Pull request merged**